vassal-app/src/main/java/VASSAL/build/widget/PieceSlot.java

```diff
@@ -101,6 +102,20 @@ public PieceSlot(PieceSlot piece) {
   public PieceSlot(CardSlot card) {
     this((PieceSlot) card);
   }
+  
+  
+  // If we're a child of a piece widget that allows scale control, get our scale from that. Otherwise default to 1.0 
+  public double getScale() {
+    Widget w = this;
+    while (w.getParent() != null) {
```

Comment:
`while ((w = w.getParent()) != null) {` saves you a call to `w.getParent()`.

Reply:
Done. At some point in my C++ past I got trained never to do assignments in comparisons :)

Reply:
There are two circumstances in which I think they're a good idea: when you're assigning to what is basically an iterator in the condition of a while loop, and when you're checking the return value of a system call for an error code.

Reply:
yeah makes sense here because it actually saves an extra call.